File: trail2/특정 구간의 합/sum-of-specific-intervals.cpp
#include <iostream>

using namespace std;

int n, m;
int arr[100];

int GetSum(int start, int end) {
    int sum = 0;
    for (int i = start - 1; i < end; i++) {
        sum += arr[i];
    }

    return sum;
}

int main() {
    cin >> n >> m;

    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for (int i = 0; i < m; i++) {
        int a1, a2;
        cin >> a1 >> a2;

        cout << GetSum(a1, a2) << endl;
    }

    // Please write your code here.

    return 0;
}

File: trail2/특정 구간의 합/sum-of-specific-intervals.py
n, m = tuple(map(int, input().split()))
a = [0] + list(map(int, input().split()))

def get_sum(a1, a2):
    sum = 0
    for i in range(a1, a2+1):
        sum += a[i]
    return sum

for i in range(m):
    a1, a2 = tuple(map(int, input().split()))
    print(get_sum(a1, a2))
